### ISSUE TITLE
[Tabs][BottomNavigation] Default badge color + text has insufficient contrast ratio

### ISSUE THREAD

**Issue by romoore** [material-components/material-components-ios#6981]
## Context
The current set of colors `UIColor.redColor` and `UIColor.whiteColor` only provide 3.95:1 contrast ratio. For the small text of the badge we need 4.5:1.  Although on iOS 10+ users can customize the color using `- UITabBarItem.badgeColor`, that's not the case for iOS 8 or iOS 9.

We should default to `MDCPalette.redPalette.tint700`, which provides 4.97:1.

## Acceptance Criteria
* Default badge background and text color have a contrast ratio of 4.5:1 or higher.

---
#### Internal data
- Associated internal bug: [b/129285783](http://b/129285783)
